app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt

```diff
@@ -543,15 +505,14 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope(), V
                     dns.add(InetAddress.getByName("1.1.1.1"))
                     dns.add(InetAddress.getByName("1.0.0.1"))
                 }.onFailure {
-                    logcat(LogPriority.WARN) { "VPN log: Error adding fallback DNS ${it.asLog()}" }
+                    logcat(WARN) { "VPN log: Error adding fallback DNS ${it.asLog()}" }
                 }
             }
         }
 
         if (!dns.containsIpv4()) {
             // never allow IPv6-only DNS
-            logcat { "VPN log: No IPv4 DNS found, return empty DNS list" }
-            return setOf()
```

Comment:
This is probably a bug (we missed removing it back when IPv6 support was addeD) that has been causing breakage. IPv6-only DNS work correctly

Reply:
In `TrackerBlockingVpnService` we filter `tunnelConfig.dns` by IPv4 - should we adjust that as well?

Reply:
Do you mean when adding the routes?

Reply:
That's right